subprojects/internal-testing/src/main/groovy/org/gradle/integtests/fixtures/TestExecutionResult.java

```diff
@@ -26,4 +26,9 @@
      * Returns the result for the given test class.
      */
     TestClassExecutionResult testClass(String testClass);
+
+    /**
+     * Returns the result for the first test class whose name starts with the given string.
+     */
+    TestClassExecutionResult testClassStartsWith(String testClass);
```

Comment:
⭕️ Instead of adding more one-off overloads I think it would be better to have `testClass(Matcher<String>)`